tangos/core/halo.py

```diff
@@ -120,12 +122,23 @@ def load(self, mode=None):
         handler this can be None or 'partial' (in a normal session) and, when running inside an MPI session,
         'server' or 'server-partial'. See https://pynbody.github.io/tangos/mpi.html.
         """
-        if self.finder_id is None:
+        halo_number = self.halo_number
+        if not hasattr(self, "finder_id"):
             finder_id = self.halo_number # backward compatibility
         else:
-            finder_id = self.finder_id 
+            if self.finder_id is None:
+                finder_id = self.halo_number
+            else:
+                finder_id = self.finder_id
+        if not hasattr(self, "catalog_index"):
+            catalog_index = self.finder_id
+        else:
+            if self.catalog_index is None:
+                catalog_index = finder_id
+            else:
+                catalog_index = self.catalog_index
```

Comment:
this is really just an attempt at backwards compatibility with a database that does not have these columns already. Maybe this is all too much and it would be better simply provide a tool for the user to update their database accordingly?

Reply:
I think simple is better here. I believe sqlalchemy takes care of adding a column of NULLs (will appear as Nones) when opening an old database. Certainly, I don't think the attribute will ever be missing